### PR TITLE
Hardening/1298 render compound simplification

### PR DESCRIPTION
**Pull request by fgalan**
This PR continues the render unification and simplification work in issue #1298. In particular:

* Domain attribute and domain metadata stuff has been removed (never implemented in NGSIv1)
* CompoundValueNode::render() removed. It is not actually needed, as the JSON generated by toJson() is exactly the same. A good side-effect of this is that apiVersion doesn't need to be passed down in the render method functions, which help to decouple a bit more NGSIv1 render code and NGSIv2 render code.

Some .test has been modified, but I think they are needed modifications and doesn't break backward compability. Please find specific comment on the .test files.

### CODE REVIEW

File: test/functionalTest/cases/0000_json_parse/jsonParseGetNotifyContext.test [hunk 1]
```diff
@@ -39,7 +39,6 @@ payload='{
         "type": "Room",
         "isPattern": "false",
         "id": "ConferenceRoom",
-        "attributeDomainName" : "ADN",
```
Comment (fgalan) [inline test/functionalTest/cases/0000_json_parse/jsonParseGetNotifyContext.test]: Obviously, attributeDomainName and metadata domain has been removed from .test.

File: test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_scope_in_v1_queries.test [hunk 2]
```diff
@@ -183,7 +183,7 @@ Date: REGEX(.*)
 05. GET entities matching q=A1==1 AND mq=A1.M1==2, see E2, using /v1/queryContext
 =================================================================================
 HTTP/1.1 200 OK
-Content-Length: 246
+Content-Length: 245
```
Comment (fgalan) [inline test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_scope_in_v1_queries.test]: This is due to a "\n" has been removed in NGSIv1 rendering. That "\n" was probably forgotten when the de-indent work in NGSIv1 was done.

File: test/functionalTest/cases/2507_csub_metadata_field/special_metadata_in_notifications_ngsiv1.test [hunk 3]
```diff
@@ -356,7 +356,7 @@ Date: REGEX(.*)
 =====================================================================================
 POST http://localhost:REGEX(\d+)/notify
```
Comment (fgalan) [inline test/functionalTest/cases/2507_csub_metadata_field/special_metadata_in_notifications_ngsiv1.test]: Changes in Content-Length in this .test due to "\n" removal in rendering logic.

File: src/lib/ngsi/Metadata.cpp [hunk 4]
```diff
@@ -284,20 +270,6 @@ std::string Metadata::render(bool comma)
...
-    //
-    // Adding newline for the types that do not use the valueTag() function
-    //
-    // FIXME: This might destroy V2 rendering
-    //   This newline is only desired for V1 requests and as this function hasn't that knowledge, we 'hardcode'
-    //   V1 behavior here, as V2 requests should use toJson and not render().
-    //   So, if V2 rendering is destroyed by this modification, it is only because the V2 rendering is using
-    //   a method that it SHOULD NOT USE !
-    //
-    out += "\n";
```
Comment (fgalan) [inline src/lib/ngsi/Metadata.cpp]: This is the "\n" that I mention in .test comment was removed.

NTC (informative)

File: src/lib/ngsi/ContextElement.cpp [hunk 5]
```diff
@@ -80,29 +80,29 @@ ContextElement::ContextElement(const std::string& id, const std::string& type, c
...
+  bool         attributeDomainNameRendered      = false; // FIXME PR
   bool         contextAttributeVectorRendered   = contextAttributeVector.size() != 0;
-  bool         domainMetadataVectorRendered     = domainMetadataVector.size() != 0;
+  bool         domainMetadataVectorRendered     = false; // FIXME PR
 
-  bool         commaAfterDomainMetadataVector   = false;  // Last element
   bool         commaAfterContextAttributeVector = domainMetadataVectorRendered;
   bool         commaAfterAttributeDomainName    = domainMetadataVectorRendered  || contextAttributeVectorRendered;
   bool         commaAfterEntityId               = commaAfterAttributeDomainName || attributeDomainNameRendered;
+#endif
```
Comment (kzangeli) [inline src/lib/ngsi/ContextElement.cpp]: All this to be removed in this PR, right?

Comment (fgalan) [reply src/lib/ngsi/ContextElement.cpp]: Yes. I missed it...

Comment (fgalan) [reply src/lib/ngsi/ContextElement.cpp]: Fixed in 6ddd2cb76fcdbc76cc452b02d631c9b1143b5b71

File: test/functionalTest/cases/0000_json_parse/jsonParseGetNotifyContext.test (hunk 1 above)
Comment (fgalan) [reply test/functionalTest/cases/0000_json_parse/jsonParseGetNotifyContext.test]: NTC

File: test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_scope_in_v1_queries.test (hunk 2 above)
Comment (fgalan) [reply test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_scope_in_v1_queries.test]: NTC

File: test/functionalTest/cases/1156_qfilters_and_compounds/qfilters_and_compounds_equals_null.test [hunk 6]
```diff
@@ -86,7 +86,7 @@ echo
 ===================================================================
 HTTP/1.1 201 Created
 Content-Length: 0
-Location: /v2/entities/E1?type=none
+Location: /v2/entities/E1?type=Thing
```
[the remark this answers is not in the document]
Comment (fgalan) [reply test/functionalTest/cases/1156_qfilters_and_compounds/qfilters_and_compounds_equals_null.test]: Fixed in 6fb965616e3fcf5b3abc286ab31e41cc7732f639

File: test/functionalTest/cases/2339_json_native_types_in_v1/json_native_types_in_v1.test [hunk 7]
```diff
@@ -155,7 +155,7 @@ Date: REGEX(.*)
 02. GET the entity using /v1/queryContext
```
[the remark this answers is not in the document]
Comment (fgalan) [reply test/functionalTest/cases/2339_json_native_types_in_v1/json_native_types_in_v1.test]: NTC

File: test/functionalTest/cases/2507_csub_metadata_field/previous_value_with_compounds_ngsiv1.test [hunk 8]
```diff
@@ -229,7 +229,7 @@ Date: REGEX(.*)
 ========================================================================
```
[the remark this answers is not in the document]
Comment (fgalan) [reply test/functionalTest/cases/2507_csub_metadata_field/previous_value_with_compounds_ngsiv1.test]: NTC

File: test/functionalTest/cases/2507_csub_metadata_field/special_metadata_in_notifications_ngsiv1.test (hunk 3 above)
Comment (fgalan) [reply test/functionalTest/cases/2507_csub_metadata_field/special_metadata_in_notifications_ngsiv1.test]: NTC

**Pull request merged**